### PR TITLE
Support activestorage

### PR DESCRIPTION
**Pull request by ksss**
activestorage adds some methods to activerecord model.

### CODE REVIEW

**Comment by ksss** [comment]
gem_rbs_collection with symlink is supported after `1.7.0.beta.1` https://github.com/ruby/rbs/pull/785 .
But, current rbs_rails's Gemfile is specifying `>= 1.6`.

Should I wait for the release of rbs v1.7.0 for this PR?

File: lib/rbs_rails/active_record.rb [hunk 1]
```diff
@@ -166,6 +167,41 @@ def reload_#{a.name}: () -> #{type_optional}
         end.join("\n")
       end
 
+      private def generated_association_methods
+        # @type var sigs: Array[String]
+        sigs = []
+
+        # Needs to require "active_storage/engine"
+        if klass.respond_to?(:attachment_reflections)
```
Comment (ksss) [inline lib/rbs_rails/active_record.rb:175]: `attachment_reflections` will be defined after "active_storage/engine" is required, and will not be defined if it is not required.

https://github.com/rails/rails/blob/3be6c2d67c0dcbefa1c252943b0188b384403ec1/activestorage/lib/active_storage/reflection.rb#L43

File: lib/rbs_rails/active_record.rb [hunk 2]
```diff
@@ -166,6 +167,41 @@ def reload_#{a.name}: () -> #{type_optional}
...
+          sigs << "module GeneratedAssociationMethods"
```
Comment (ksss) [inline lib/rbs_rails/active_record.rb:176]: This is a tracing of the rails module.

https://github.com/rails/rails/blob/18707ab17fa492eb25ad2e8f9818a320dc20b823/activestorage/lib/active_storage/attached/model.rb#L53
https://github.com/rails/rails/blob/4ae0390c1051a39ec16a4a95451cd06172a1b260/activerecord/lib/active_record/core.rb#L368-L376

File: lib/rbs_rails/active_record.rb [hunk 3]
```diff
@@ -366,8 +404,16 @@ def authenticate_#{attribute}: (String) -> (#{klass_name} | false)
...
 
           args = args_to_type(body_node.children[1])
-          "def #{singleton ? 'self.' : ''}#{name}: #{args} -> #{relation_class_name}"
-        end.compact.join("\n")
+          "def #{prefix}#{name}: #{args} -> #{relation_class_name}"
+        end.compact
+
+        if klass.respond_to?(:attachment_reflections)
+          klass.attachment_reflections.each do |name, _reflection|
+            sigs << "def #{prefix}with_attached_#{name}: () -> #{relation_class_name}"
```
Comment (ksss) [inline lib/rbs_rails/active_record.rb:412]: https://github.com/rails/rails/blob/18707ab17fa492eb25ad2e8f9818a320dc20b823/activestorage/lib/active_storage/attached/model.rb#L73

File: test/expectations/user.rbs [hunk 4]
```diff
@@ -184,6 +184,30 @@ class User < ApplicationRecord
...
+  def build_avatar_attachment: (untyped) -> ActiveStorage::Attachment
+  def create_avatar_attachment: (untyped) -> ActiveStorage::Attachment
+  def create_avatar_attachment!: (untyped) -> ActiveStorage::Attachment
+  def reload_avatar_attachment: () -> ActiveStorage::Attachment?
+  def avatar_blob: () -> ActiveStorage::Blob?
+  def avatar_blob=: (ActiveStorage::Blob?) -> ActiveStorage::Blob?
+  def build_avatar_blob: (untyped) -> ActiveStorage::Blob
+  def create_avatar_blob: (untyped) -> ActiveStorage::Blob
+  def create_avatar_blob!: (untyped) -> ActiveStorage::Blob
+  def reload_avatar_blob: () -> ActiveStorage::Blob?
```
Comment (ksss) [inline test/expectations/user.rbs:198]: https://github.com/rails/rails/blob/18707ab17fa492eb25ad2e8f9818a320dc20b823/activestorage/lib/active_storage/attached/model.rb#L70-L71

**Comment by ksss** [comment]
Rebased and use rbs 1.7.0.

But I have not been able to solve the CI problem.

> nokogiri-1.12.5-x86_64-linux requires ruby version < 3.1.dev, >= 2.5, which is
> incompatible with the current version, ruby 3.1.0p-1

CI seems to be down with the following problems.
https://github.com/sparklemotion/nokogiri/issues/2185

**Comment by ksss** [comment]
I resolved CI problem!
I simply avoided using the precompiled gem in the head.

**Review by pocke — APPROVED** [review]
Thanks! 🚀

**Comment by pocke** [comment]
I've reverted some commits because they are no longer needed on the master branch. And I've merged master into this PR to check it runs on master branch.

**Comment by ksss** [comment]
Thank you so much ✨

**Pull request merged**